core/templates/dev/head/pages/state_editor/StateContentEditorDirective.js

```diff
@@ -53,6 +53,22 @@ oppia.directive('stateContentEditor', [
 
           $scope.contentEditorIsOpen = false;
           $scope.isEditable = EditabilityService.isEditable;
+          $scope.isCardHeightLimitReached = false;
+          $scope.showCardHeightLimitWarning = true;
+
+          var cardHeightWatcher = $scope.$watch(function() {
```

Comment:
Actually I'm not sure about this. Wouldn't it be simpler to just have:

$scope.isCardHeightLimitReached = function() {
  ...
  return true/false;
}

(note that in this case isCardHeightLimitReached is a function, so it's fine for the name to start with "is") and then do `ng-if="isCardHeightLimitReached() && showCardHeightLimitWarning">`? In general we try to avoid defining custom $watch calls (and in fact we try to avoid using $watch at all) since it leads to performance issues.

Reply:
Should I try some other alternative to `$watch`?

Reply:
Would the suggestion I proposed eliminate the need for $watch?

Reply:
No, because we need to monitor the height of  dummy card while user is typing. Once the height exceeds 630px, we need to provide a warning message. If the user deletes a few lines and height becomes less than 630px, the warning message has to be removed. In order to monitor the height of dummy card continuously, $watch is required here.

Reply:
But I think ng-if="function()" has an inbuilt watcher, right? I.e. it still gets updated in real-time, because the "watching" happens behind the scenes (but in this case it's wrapped behind standard Angular constructs).

Reply:
Ok, got it! Made the changes, PTAL.